arch/arm64/core/cache.c

```diff
@@ -57,3 +73,67 @@ size_t arch_dcache_line_size_get(void)
 
 	return dcache_line_size;
 }
+
+/*
+ * operation for all data cache
+ * ops:  K_CACHE_INVD: invalidate
+ *	 K_CACHE_WB: clean
+ *	 K_CACHE_WB_INVD: clean and invalidate
+ */
+int arch_dcache_all(int op)
+{
+	uint32_t clidr_el1, csselr_el1, ccsidr_el1;
+	uint8_t loc, ctype, cache_level, line_size, way_pos;
+	uint32_t max_ways, max_sets, dc_val, set, way;
+
+	if (op != K_CACHE_INVD && op != K_CACHE_WB && op != K_CACHE_WB_INVD)
+		return -ENOTSUP;
+
+	clidr_el1 = read_clidr_el1();
+
+	loc = (clidr_el1 >> CLIDR_EL1_LOC_SHIFT) & CLIDR_EL1_LOC_MASK;
+	if (!loc)
+		return 0;
+
+	for (cache_level = 0; cache_level < loc; cache_level++) {
+		ctype = (clidr_el1 >> cache_level * 3) & BIT_MASK(3);
```

Comment:
defines for this as well?

Reply:
Done, but I can't update this pull request and re-open it, so I create a new one https://github.com/zephyrproject-rtos/zephyr/pull/34525, please continue to review, thanks.